stablehlo/reference/Types.cpp

```diff
@@ -36,8 +36,10 @@ bool isSupportedSignedIntegerType(Type type) {
          type.isSignlessInteger(64);
 }
 
+bool isSupportedBooleanType(Type type) { return type.isSignlessInteger(1); }
+
 bool isSupportedIntegerType(Type type) {
-  return isSupportedUnsignedIntegerType(type) ||
+  return isSupportedBooleanType(type) || isSupportedUnsignedIntegerType(type) ||
```

Comment:
I don't think we should be treating booleans as integers. 

On the plus side, this allows us to avoid changing `Element.h` from `std::variant<APInt, APFloat, std::pair<APFloat, APFloat>>` to `std::variant<bool, APInt, APFloat, std::pair<APFloat, APFloat>>` and updating `map` in `Element.cpp`.

On the minus side, this hides logical errors. E.g. a bunch of existing code e.g. overloaded operators in `Element` will silently accept booleans, and some of that code, e.g. `Element::operator-` will be silently incorrect.

Given that, I recommend that we model booleans as a separate family of types and as a separate variant in `Element.h`. Existing operators in `Element` can either be updated in this PR, or can `report_fatal_error` for now and wait until #285.

Reply:
done.